crates/loro-internal/src/state.rs

```diff
@@ -40,6 +42,9 @@ pub struct DocState {
     pub(super) states: FxHashMap<ContainerIdx, State>,
     pub(super) arena: SharedArena,
 
+    // resolve event stuff
+    weak_state: Weak<Mutex<DocState>>,
+    global_txn: Weak<Mutex<Option<Transaction>>>,
```

Comment:
Maybe we can make `DocState` as the owner of `global_txn`, and remove the `global_txn` in `Loro`.